aderyn_core/src/ast/source_units.rs

```diff
@@ -8,6 +8,7 @@ use std::{collections::HashMap, io};
 #[derive(Clone, Debug, Deserialize, Eq, Serialize, PartialEq, Hash)]
 #[serde(untagged)]
 pub enum SourceUnitNode {
+    FunctionDefinition(FunctionDefinition),
```

Comment:
This is strategically placed at the top (or I should say above ErrorDefinition to be specific) because its' fields are a superset of ErrorDefinition and therefore if Rust can deserialize the object into this it will do so. Otherwise, ErrorDefinition would be the first satisfied case and would stop the program from detecting free functions